public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -97,15 +97,30 @@ export class PrometheusDatasource {
       activeTargets.push(target);
 
       var query: any = {};
-      query.expr = this.templateSrv.replace(target.expr, options.scopedVars, self.interpolateQueryExpr);
-      query.requestId = options.panelId + target.refId;
       query.instant = target.instant;
```

Comment:
The code within this loop should be extracted into a new function.

Reply:
Done, function named createQuery().